metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -738,11 +738,17 @@ def get_all_databases_and_tables(
         self,
     ) -> Tuple[List[Mapping[str, Any]], List[Dict]]:
         all_databases = [*self.get_all_databases()]
-        all_tables = [
-            tables
-            for database in all_databases
-            for tables in self.get_tables_from_database(database)
-        ]
+        all_tables = []
+        for database in all_databases:
+            try:
+                for tables in self.get_tables_from_database(database):
+                    all_tables.append(tables)
+            except Exception as e:
+                self.report.failure(
```

Comment:
Do we know what kinds of exceptions are to happen here. Trying to determine if there are any that need to be reported as warnings instead.

Reply:
Access denied causing the whole ingestion to fail

Reply:
This `self.report.failure` would cause ingestion to fail but will still do best effort metadata extraction.

If possible, it will be good to identify when its permission error and use more instructive and actionable message when reporting failure.

Reply:
+1 I agree. This should have had been error reporting in the first place!